apps/src/templates/Dialog.jsx

```diff
@@ -138,8 +141,24 @@ const Dialog = React.createClass({
     onCancel: whenNoChildOfTypes(Buttons),
     confirmText: whenNoChildOfTypes(Buttons),
     onConfirm: whenNoChildOfTypes(Buttons),
+    confirmType: whenNoChildOfTypes(Buttons),
   }),
 
+  handleKeyDown(event) {
+    // Focus the next button when tab is pressed, to prevent the
+    // user from selecting elements outside of the dialog.
+    if (event.key === 'Tab') {
+      const buttons = this.refs.baseDialog.getButtons();
+      if (buttons.length) {
+        // Focus the next button, or the first button if none is focused.
+        const curIndex = buttons.findIndex(btn => btn === document.activeElement);
+        const nextIndex = (curIndex + 1) % buttons.length;
+        buttons[nextIndex].focus();
+      }
+      event.preventDefault();
```

Comment:
If we do this for inputs, do we also want to do it for links? I feel like we are starting to reinvent the wheel here.